management/vmutils/rolestate.go

```diff
@@ -0,0 +1,31 @@
+package vmutils
+
+import (
+	"time"
+
+	"github.com/MSOpenTech/azure-sdk-for-go/management"
+	vm "github.com/MSOpenTech/azure-sdk-for-go/management/virtualmachine"
+)
+
+// Blocks until all role instances in deployment to reach desired power state
+func WaitForDeploymentPowerState(client management.Client, cloudServiceName, deploymentName string, desiredPowerstate vm.PowerState) error {

```

Comment:
RoleStatus can be 'StoppedVM' with PowerState still 'Started' until the VM is finally turned off. I have no idea why there are 2 statuses, but experimentally, I've come to trust PowerState. Since there seem to be no other users, I serve myself... :smile: